tools/site_admin/manage_user_sessions.php

```diff
@@ -0,0 +1,109 @@
+<?php
+$relPath = "./../../pinc/";
+include_once($relPath.'base.inc');
+include_once($relPath.'theme.inc');
+include_once($relPath.'User.inc');
+
+require_login();
+
+if (!user_is_a_sitemanager()) {
+    die(_("You are not authorized to invoke this script."));
+}
+
+// --------------------------------------
+
+$username = $_POST['username'] ?? ($_GET['username'] ?? null);
+$action = get_enumerated_param($_POST, 'action', null, ['logout'], true);
+
+$title = _("Manage User Sessions");
+output_header($title, NO_STATSBAR);
+
+echo "<h1>$title</h1>\n";
+echo "<p>" . _("This page shows how many separate sessions (logins) a user has and the option of logging them out.") . "</p>";
+
+show_username_form($username);
+
+if ($username) {
+    try {
+        $user = new User($username);
+    } catch (NonexistentUserException $exception) {
+        echo "<p class='error'>" . _("Invalid username") . "</p>";
+        exit;
+    }
+
+    echo "<hr>";
+    echo "<h2>$username ($user->real_name)</h2>";
+
+    if ($action == "logout") {
+        delete_sessions($username);
+    } else {
+        show_sessions_form($username);
+    }
+}
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+function show_username_form(?string $username): void
+{
+    echo "<form method='GET'>\n";
+    echo "<p>" . _("Username") . ": ";
+    echo "<input name='username' type='text' value='$username' size='20' autocapitalize='none' required>\n";
```

Comment:
do we need to encode username here?

Reply:
Yes, we should. And this was a copy/pasta from another script. I've pushed up two commits (one will be squashed in a rebase later) that correctly escapes this and pulls the function into a separate file we can use here, in the original, and one other place in the `site_admin/*`.